src/cpp/rtps/builtin/discovery/participant/PDP.h

```diff
@@ -129,7 +129,7 @@ class PDP : public fastdds::statistics::rtps::IProxyQueryable
      *
      * @return true if enabled correctly, or if already enabled; false otherwise
      */
-    bool enable();
+    virtual bool enable();
```

Comment:
Instead of making this virtual, I would put the special actions in a `virtual void pre_enable_actions()` and call that [here](https://github.com/eProsima/Fast-DDS/blob/233434fffd030a853b0dfa5228e5c790e451cab7/src/cpp/rtps/builtin/discovery/participant/PDP.cpp#L415).

The new method will have an empty implementation in PDP, and the backup restore in PDPServer.